File: symmetry.py
#Time Complexity: O(N)
#Space Complexity: O(N)

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from queue import Queue
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if root==None:
            return True
        q=Queue()
        q.put(root.left)
        q.put(root.right)

        while not(q.empty()):
            left=q.get()
            right=q.get()
            
            if left==None and right==None:
                continue
            if left==None or right==None:
                return False
            if left.val !=right.val:
                return False
            q.put(left.left)
            q.put(right.right)
            q.put(left.right)
            q.put(right.left)

        return True

File: symmetry.java
// Time Complexity : o(n)
// Space Complexity : o(n)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no


// Your code here along with comments explaining your approach

class Solution {
    public boolean isSymmetric(TreeNode root) {
        
       return  root==null || findSym(root.left, root.right);
        
    }
    public boolean findSym(TreeNode right, TreeNode left){
 
        if(left==null || right==null)
        return left==right;
        if(left.val!=right.val)
            return false;
    return findSym(left.left, right.right) && findSym(left.right, right.left);
}
            
        }

    
